kafka-connect-s3/src/test/java/io/confluent/connect/s3/S3SinkConnectorTest.java

```diff
@@ -38,5 +64,21 @@ public void connectorType() {
     Connector connector = new S3SinkConnector();
     assertTrue(SinkConnector.class.isAssignableFrom(connector.getClass()));
   }
+
+  @Test(expected = ConnectException.class)
+  public void checkNonExistentBucket() throws Exception {
+    configProps.put(S3SinkConnectorConfig.S3_BUCKET_CONFIG, "non-existent-bucket");
+    connector = spy(new S3SinkConnector(new S3SinkConnectorConfig(configProps)));
+    doReturn(false).when(connector, "checkBucketExists", Mockito.any());
```

Comment:
See if it is possible to avoid mocking a private method like so. I think if this is moved to a common lib method, then there will be no need to mock private method. Same applies for the other reverse function. You could then use EasyMock.